### ISSUE TITLE
datepicker HostListener mouse issue v27

### ISSUE THREAD

**Issue by scttcper** [ng-bootstrap/ng-bootstrap#1660]
Having a `@HostListener('mouseup')` or `@HostListener('window:mouseup')` breaks datepicker. I can't tell yet what's doing it but it seems to be redrawing the days on every event. Having a `@HostListener('mousemove')` causes similar redraws but at least it still works when the mouse isn't moving. The HostListener can be on any parent component and still causes the same issue. I don't believe this was an issue in v26

Try clicking a date on the calendar.
http://plnkr.co/edit/wC4XeI8OOIBAMLo9yeFc?p=preview

**Comment by pkozlowski-opensource** [comment]
@scttcper I can't see any host listener in the linked plunker... Could you please update the plunker so it actually shows the issue you are facing?

**Comment by scttcper** [comment]
@pkozlowski-opensource its there in app.ts

```typescript
@HostListener('window:mouseup')
 doSomething() {
 }
```

mousemove causes same thing
```typescript
@HostListener('window:mousemove')
 mousemover() {
 }
```

**Comment by maxokorokov** [comment]
This most likely was caused by https://github.com/ng-bootstrap/ng-bootstrap/commit/5cfd9e43a3a9d06a29fefce4e468075c9b58effa#diff-e5d0a2ff3eb57507d9f2efd1bce31f54R39

And should be fixed as soon as #1551 lands

**Comment by coldhamix** [comment]
I am experiencing the same issue. I have a custom outside-click

**Comment by coldhamix** [comment]
I am experiencing the same issue when having a datepicker in a popover which internally has a `@HostListener('document:mousedown')`. 
Clicking on a date does nothing in v27, but in v25 everything works just fine.

**Comment by pkozlowski-opensource** [comment]
Fixed via #1551

**Comment by scttcper** [comment]
Thanks @maxokorokov @pkozlowski-opensource this is fixed on 28 🦄
